### PR TITLE
LMSDEMO - 27 - add style to download anchor

### PR DESCRIPTION
**Pull request by postrowinski**

### CODE REVIEW

File: src/components/primitives/Download/Download.md [hunk 1]
```diff
@@ -13,6 +14,7 @@ import img1 from "./Download.png";
   <Download
     href="https://escolalms.github.io/headless-format/paper.pdf"
     fileName={<strong>myPaper.pdf</strong>}
+    darkTheme
```
Comment (qunabu) [inline src/components/primitives/Download/Download.md:17]: https://github.com/EscolaLMS/Components/blob/main/src/theme/provider.tsx#L11 

check theme `mode` attribute to see if we're in dark or light mode

File: src/components/primitives/Download/Download.tsx [hunk 2]
```diff
@@ -1,42 +1,68 @@
 import * as React from "react";
-
-import styled, { withTheme } from "styled-components";
+import styled, { ThemeContext, withTheme } from "styled-components";
+import { getFontFromTheme } from "../../../theme/provider";
 
 export interface ComponentProps {
   href: string;
   fileName?: string | React.ReactNode;
+  darkTheme?: boolean;
```
Comment (qunabu) [inline src/components/primitives/Download/Download.tsx:8]: see above

File: src/components/primitives/Download/Download.tsx [hunk 3]
```diff
@@ -1,42 +1,68 @@
...
-  flex-wrap: nowrap;
-  justify-content: space-between;
+  font-size: 1.1em;
+  diplay: inline-block;
+  width: 300px;
+  margin: 10px;
+  font-family: ${(props) => getFontFromTheme(props.theme).fontFamily};
+  .download {
+    background: ${(props) => {
+      return props.darkTheme ? props.theme.body.gray1 : props.theme.body.white;
```
Comment (qunabu) [inline src/components/primitives/Download/Download.tsx:19]:
```suggestion
      return props.theme.mode !== 'light' ? props.theme.body.gray1 : props.theme.body.white;
```

File: src/components/primitives/Download/Download.tsx [hunk 4]
```diff
@@ -1,42 +1,68 @@
...
+    }};
+    color: ${(props) => {
+      return props.darkTheme ? props.theme.body.white : props.theme.body.black;
```
Comment (qunabu) [inline src/components/primitives/Download/Download.tsx:22]:
```suggestion
      return props.theme.mode !== 'light' ? props.theme.body.white : props.theme.body.black;
```

File: src/components/primitives/Download/Download.tsx [hunk 5]
```diff
@@ -1,42 +1,68 @@
...
+    display: flex;
+    align-items: center;
+    justify-content: space-between;
+    curson: pointer;
+    transition: all 0.3s;
+    &:hover {
+      background: ${(props) => {
+        const darkHover = props.theme.body.gray2;
+        const ligthHover = props.theme.body.gray3;
+        return props.darkTheme ? darkHover : ligthHover
```
Comment (qunabu) [inline src/components/primitives/Download/Download.tsx:35]:
```suggestion
        return props.theme.mode !== 'light' ?  props.theme.body.gray2 : props.theme.body.gray3
```

File: src/components/primitives/Download/Download.tsx [hunk 6]
```diff
@@ -1,42 +1,68 @@
...
+  const theme = React.useContext(ThemeContext);
 
   const name = React.useMemo(() => {
     return fileName || href.split("/").pop();
   }, [href, fileName]);
 
   return (
     <StyledDiv {...props}>
-      <a download href={href} target="_blank">
+      <a className="download" download href={href} target="_blank">
```
Comment (qunabu) [inline src/components/primitives/Download/Download.tsx:54]: imho its better to use more precise classes link `download-btn` or something like that

File: src/components/primitives/Download/Download.tsx [hunk 7]
```diff
@@ -1,42 +1,68 @@
...
-              fill="#4A4A4A"
+              fill={props.darkTheme ? theme.body.white : theme.body.black}
```
Comment (qunabu) [inline src/components/primitives/Download/Download.tsx:65]:
```suggestion
              fill={props.theme.mode !== 'light' ? theme.body.white : theme.body.black}
```

File: src/theme/provider.tsx [hunk 8]
```diff
@@ -4,13 +4,27 @@ import React, { useState, useEffect, useCallback } from "react";
...
+  white: string,
+  gray5: string,
+  gray4: string,
+  gray3: string,
+  gray2: string,
+  gray1: string,
+  black: string;
+}
+
+export interface SharedDefaultTheme {
```
Comment (qunabu) [inline src/theme/provider.tsx:17]: all themes **should** share same attributes, no sure why this is needed this

File: src/components/primitives/Download/Download.md [hunk 9]
```diff
@@ -2,6 +2,7 @@
 import { GlobalThemeProvider } from "../../../theme/provider";
 import ImageModal from "../../../styleguide/ImageModal";
 import img1 from "./Download.png";
+import Download from "./Download";
 
 <GlobalThemeProvider>
```
Comment (qunabu) [inline src/components/primitives/Download/Download.md:7]: the description should display a component in all possible themes (see other examples) 


```tsx
import { DefaultTheme, ThemeProvider } from "styled-components";
import themes from "../../../theme";


<React.Fragment>
  {Object.entries(themes).map((theme) => (
    <ThemeProvider theme={theme[1]} key={theme[0]}>
        { // your component here }
    </ThemeProvider>
)}
</React.Fragment>
```

**Review by qunabu — CHANGES_REQUESTED** [review]
see the review in each file comment

File: src/components/primitives/Download/Download.md (hunk 9 above)
Comment (postrowinski) [reply src/components/primitives/Download/Download.md:7]: by in this case Download component has only mode type (light and dark), there is no difference in theme

File: src/theme/provider.tsx (hunk 8 above)
Comment (postrowinski) [reply src/theme/provider.tsx:17]: they share all attribute, by some attributes has the same values in each theme, for example black, white and gray palette colors.

**Review by qunabu — APPROVED** [review]

**Pull request merged**